packages/mask/src/components/shared/ApplicationBoard.tsx

```diff
@@ -154,5 +170,89 @@ interface RenderEntryComponentWrapperProps {
 
 function RenderEntryComponentWrapper({ application }: RenderEntryComponentWrapperProps) {
     const RenderEntryComponent = application.entry.RenderEntryComponent!
-    return <RenderEntryComponent disabled={!application.enabled} />
+    return application.entry.nextIdRequired ? (
+        <RenderEntryComponentWithNextIdRequired application={application} />
+    ) : (
+        <RenderEntryComponent disabled={!application.enabled} />
+    )
+}
+
+function RenderEntryComponentWithNextIdRequired({ application }: RenderEntryComponentWrapperProps) {
+    const ui = activatedSocialNetworkUI
+    const { t } = useI18N()
+    const platform = ui.configuration.nextIDConfig?.platform as NextIDPlatform
+    const lastStateRef = currentSetupGuideStatus[ui.networkIdentifier]
+    const lastState_ = useValueRef(lastStateRef)
+    const lastState = useMemo<SetupGuideCrossContextStatus>(() => {
+        try {
+            return JSON.parse(lastState_)
+        } catch {
+            return {}
+        }
+    }, [lastState_])
+    const lastRecognized = useLastRecognizedIdentity()
+    const getUsername = () =>
+        lastState.username || (lastRecognized.identifier.isUnknown ? '' : lastRecognized.identifier.userId)
+    const [username] = useState(getUsername)
```

Comment:
This is confusing for me. If you don't need `setState`, why not use `useMemo`.

Reply:
done